pkg/modelregistry/jsonvalues/convertJson_test.go

```diff
@@ -0,0 +1,166 @@
+	assert.Equal(t, len(readOnlyPaths), 37)
+
+	val01 := change.ConfigValue{Path: "/openconfig-system:system/ntp/state/auth-mismatch",
+		TypedValue: *change.CreateTypedValueFloat(123456.00000)}
+	val02 := change.ConfigValue{Path: "/openconfig-system:system/ntp/state/enable-ntp-auth",
+		TypedValue: *change.CreateTypedValueBool(true)}
+	val03 := change.ConfigValue{Path: "/openconfig-system:system/ntp/state/ntp-source-address",
+		TypedValue: *change.CreateTypedValueString("192.168.0.17")}
+	val04 := change.ConfigValue{Path: "/openconfig-interfaces:interfaces/interface[1]/state/mtu",
+		TypedValue: *change.CreateTypedValueFloat(960.0)}
+	val05 := change.ConfigValue{Path: "/openconfig-interfaces:interfaces/interface[1]/name",
+		TypedValue: *change.CreateTypedValueString("eth1")}
+	val06 := change.ConfigValue{Path: "/openconfig-interfaces:interfaces/interface[1]/state/description",
+		TypedValue: *change.CreateTypedValueString("new if desc")}
+	val07 := change.ConfigValue{Path: "/openconfig-interfaces:interfaces/interface[1]/subinterfaces/subinterface[15]/state/ifindex",
+		TypedValue: *change.CreateTypedValueFloat(10.0)}
+	val08 := change.ConfigValue{Path: "/openconfig-interfaces:interfaces/interface[1]/subinterfaces/subinterface[15]/index",
+		TypedValue: *change.CreateTypedValueString("120")}
+	val09 := change.ConfigValue{Path: "/openconfig-interfaces:interfaces/interface[1]/subinterfaces/subinterface[15]/state/admin-status",
+		TypedValue: *change.CreateTypedValueString("UP")}
+	val10 := change.ConfigValue{Path: "/openconfig-system:system/openconfig-aaa:aaa/server-groups/server-group[10]/servers/server[199]/state/connection-aborts",
+		TypedValue: *change.CreateTypedValueFloat(10.0)}
+	val11 := change.ConfigValue{Path: "/openconfig-system:system/openconfig-aaa:aaa/server-groups/server-group[10]/name",
+		TypedValue: *change.CreateTypedValueString("g1")}
+	val12 := change.ConfigValue{Path: "/openconfig-system:system/openconfig-aaa:aaa/server-groups/server-group[10]/servers/server[199]/address",
+		TypedValue: *change.CreateTypedValueString("192.168.0.7")}
+	//jsonPathValues := []*change.ConfigValue{&val01, &val02, &val03, &val04, &val05}
+	jsonPathValues := []*change.ConfigValue{&val01, &val02, &val03, &val04,
+		&val05, &val06, &val07, &val08, &val09, &val10, &val11, &val12}
+
+	correctedPathValues, err := correctJSONPaths(jsonPathValues, readOnlyPaths)
+	assert.NilError(t, err)
+
+	for _, v := range correctedPathValues {
+		fmt.Printf("%s %v\n", (*v).Path, v.String())
+	}
+	assert.Equal(t, len(correctedPathValues), 8)
+
+	for _, correctedPathValue := range correctedPathValues {
+		switch correctedPathValue.Path {
+		case "/openconfig-system:system/ntp/state/auth-mismatch":
+			assert.Equal(t, correctedPathValue.Type, change.ValueTypeUINT)
+			assert.Equal(t, len(correctedPathValue.TypeOpts), 0)
+			assert.Equal(t, correctedPathValue.String(), "123456")
```

Comment:
is the value worth extracting and reusing during construction and test ?

Reply:
done